### PR TITLE
simplify endpoint inputs

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR does:
- update the endpoint input
   -  replace `z` by `group` parameter in `/tilejson` endpoint
   - add `group` in `/variables` and `/info` endpoint 
   - reorder parameters 
- update variable name to make a difference between xarray.dataset and xarray.dataarray 

This will break any deployed Frontend

### CODE REVIEW

File: titiler/xarray/factory.py [hunk 1]
```diff
@@ -34,8 +33,8 @@ def xarray_open_dataset(
         "decode_times": decode_times,
         "consolidated": True,
     }
-    if multiscale:
-        xr_open_args["group"] = z
+    if group:
+        xr_open_args["group"] = group
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:37]: we don't need to have `group` and `multiscale` options, if a `group` is set we will use it

File: titiler/xarray/factory.py [hunk 2]
```diff
@@ -49,47 +48,48 @@ def xarray_open_dataset(
...
-) -> xarray.Dataset:
-    """Update dataset."""
-    src = src.rename({"lat": "y", "lon": "x"})
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray."""
+    ds = ds.rename({"lat": "y", "lon": "x"})
     if drop_dim:
         dim_to_drop, dim_val = drop_dim.split("=")
-        src = src.sel({dim_to_drop: dim_val}).drop(dim_to_drop)
+        ds = ds.sel({dim_to_drop: dim_val}).drop(dim_to_drop)
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:61]: This could be in the `xarray_open_dataset` function 🤷

File: titiler/xarray/factory.py [hunk 3]
```diff
@@ -49,47 +48,48 @@ def xarray_open_dataset(
...
 
-    ds = src[variable]
+    da = ds[variable]
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:63]: clear split between dataset and dataarray

File: titiler/xarray/factory.py [hunk 4]
```diff
@@ -49,47 +48,48 @@ def xarray_open_dataset(
...
-    crs = ds.rio.crs or "epsg:4326"
-    ds.rio.write_crs(crs, inplace=True)
+    crs = da.rio.crs or "epsg:4326"
+    da.rio.write_crs(crs, inplace=True)
 
     # TODO - address this time_slice issue
-    if "time" in ds.dims:
-        times = [str(x.data) for x in ds.time]
+    if "time" in da.dims:
+        # times = [str(x.data) for x in ds.time]
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:78]:
```suggestion
```

File: titiler/xarray/factory.py [hunk 5]
```diff
@@ -295,8 +300,27 @@ def tilejson_endpoint(  # type: ignore
                 self.default_tms,
                 description=f"TileMatrixSet Name (default: '{self.default_tms}')",
             ),
-            src_path: str = Depends(self.path_dependency),
+            url: str = Query(..., description="Dataset URL"),
+            group: Optional[int] = Query(
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:304]: `z` -> `group`

**Comment by abarciauskas-bgse** [comment]
LGTM @vincentsarago noted it will break front ends using it, when I deploy I will modify as necessary zarr-viz-explorations.

I don't think https://github.com/NASA-IMPACT/veda-ui/pull/538/files needs to change.

**Pull request merged**